transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -78,9 +77,13 @@ private boolean add0(ByteBuf buf, int index, int len, InetSocketAddress recipien
         return true;
     }
 
-    @Override
-    public boolean processMessage(Object msg) {
-        return msg instanceof DatagramPacket && addReadable((DatagramPacket) msg);
+    void add(ChannelOutboundBuffer buffer, final InetSocketAddress remote) throws Exception {
```

Comment:
@normanmaurer if I understand correctly, it would be better here to not pass a separate address but just a `boolean` to indicate whether connected, and if so override the remote address to be `null` in `processMessage(...)`.

This would save overhead of copying the address for every packet in `NativeDatagramPacket#init(...)`. You can send NULL address when connected, I think that also applies to `sendmmsg`

Reply:
let me try if this works... good point.